src/System.Private.CoreLib/shared/System/Array.cs

```diff
@@ -1591,10 +1582,10 @@ public static bool TrueForAll<T>(T[] array, Predicate<T> match)
         private readonly struct SorterObjectArray
         {
             private readonly object[] keys;
-            private readonly object[] items;
+            private readonly object[]? items;
             private readonly IComparer comparer;
 
-            internal SorterObjectArray(object[] keys, object[] items, IComparer comparer)
+            internal SorterObjectArray(object[] keys, object[]? items, IComparer comparer)
```

Comment:
Should items be `object?[]?` ?

Reply:
If it's `object?[]` then any call sites that pass `object[]` warn, because it believes the implementation might store a null into something that's not supposed to have nulls; `object[]` is the more permissive type from the caller's perspective, because either `object?[]` or `object[]` could be passed into it.

Reply:
Actually, my explanation here may be bogus.  I'll take a deeper look at this and fix it if necessary.  It's internal though so it doesn't matter all that much.

Reply:
I did it this way because we don't actually care whether items contains null nor do we assign any new nulls into it... since we're sorting, we're just moving around the existing contents.  It's also internal so it's not a big deal.  But I'll take another look at it to see whether it'd be better as object?[]? instead.

Reply:
Since it’s internal it is not a big deal I just got confused at looking at the implementation actually assigning nulls and accepting nulls, so wanted to understand if it was a magic trick to avoid a ton of warnings. Thanks!